### PR TITLE
[#2] hrw: add typed methods for hashers

### PR DESCRIPTION
**Pull request by dstepanov-yadro**
Add generic methods for sort-methods used in node

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,8 +1,13 @@
 module github.com/TrueCloudLab/hrw
 
-go 1.14
```
Comment (fyrchik) [inline go.mod:3]: Worth a separate commit.

File: hrw.go [hunk 2]
```diff
@@ -83,6 +83,17 @@ func SortSliceByValue(slice interface{}, hash uint64) {
 	}
 }
 
+// SortHasherSliceByValue received []Hasher and hash to sort by value-distance
```
Comment (fyrchik) [inline hrw.go]: s/received/receives/
dot in the end of the comment: there is a linter which complains + Godoc is nicer.

**Review by fyrchik — COMMENTED** [review]
Could you also add `benchstat` output (or just benchmark result) in the commit message?

File: go.mod (hunk 1 above)
Comment (dstepanov-yadro) [reply go.mod:3]: fixed

File: hrw.go (hunk 2 above)
Comment (dstepanov-yadro) [reply hrw.go]: fixed

File: README.md [hunk 3]
```diff
@@ -14,25 +14,37 @@
...
-BenchmarkSort_fnv_100-8                           300000              5261 ns/op            1856 B/op          3 allocs/op
-BenchmarkSort_fnv_1000-8                           10000            119462 ns/op           16448 B/op          3 allocs/op
-BenchmarkSortByIndex_fnv_10-8                    3000000               546 ns/op             384 B/op          7 allocs/op
-BenchmarkSortByIndex_fnv_100-8                    200000              5965 ns/op            2928 B/op          7 allocs/op
-BenchmarkSortByIndex_fnv_1000-8                    10000            127732 ns/op           25728 B/op          7 allocs/op
-BenchmarkSortByValue_fnv_10-8                    2000000               962 ns/op             544 B/op         17 allocs/op
-BenchmarkSortByValue_fnv_100-8                    200000              9604 ns/op            4528 B/op        107 allocs/op
-BenchmarkSortByValue_fnv_1000-8                    10000            111741 ns/op           41728 B/op       1007 allocs/op
+BBenchmarkSort_fnv_10-8                                           4812801               240.9 ns/op           216 B/op          4 allocs/op
```
Comment (fyrchik) [inline README.md]: `BB`?

**Review by fyrchik — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (fyrchik) [reply README.md]: Frankly, I would rather delete this from README -- raw numbers, without any explanation or meaningful implications. Or, maybe at least add CPU model/RAM capacity. Like in this diff: the numbers have changed even though for most of the benchmarks the code is the same.

Not in this PR, of course. cc @alexvanin

**Review by carpawell — COMMENTED** [review]
What does the last commit mean? Also, agree with @fyrchik that comparing (before/after on the same machine) results somewhere (e.g. commit msg) would be nice.

**Comment by dstepanov-yadro** [comment]
> What does the last commit mean? Also, agree with @fyrchik that comparing (before/after on the same machine) results somewhere (e.g. commit msg) would be nice.

git command typo, fixed.

There is no fixes in existed methods, so no diff in existed benchmarks.

**Review by alexvanin — APPROVED** [review]

**Pull request merged**